ksqldb-parser/src/test/java/io/confluent/ksql/parser/AstBuilderTest.java

```diff
@@ -206,6 +208,41 @@ public void shouldExtractAsAliasedJoinDataSources() {
     assertThat((Join) result.getFrom(), hasRights(new AliasedRelation(TEST2, SourceName.of("T2"))));
   }
 
+  @Test
+  public void shouldHandleSelectStructAll() {
+    // Given:
+    final SingleStatementContext stmt =
+        givenQuery("SELECT NESTED_ORDER_COL->* FROM NESTED_STREAM;");
+
+    // When:
+    final Query result = (Query) builder.buildStatement(stmt);
+
+    // Then:
+    assertThat(result.getSelect(), is(new Select(ImmutableList.of(
+        new StructAll(column("NESTED_ORDER_COL"))
+    ))));
+  }
+
+  @Test
+  public void shouldHandleSelectStructAllOnNestedStruct() {
```

Comment:
Ok, so this is the class where we can run through the various cases we want to make sure and cover.  (Like arrays, maps, functions, whatever...)